packages/shorebird_cli/lib/src/artifact_manager.dart

```diff
@@ -282,6 +282,15 @@ class ArtifactManager {
     return archiveDirectory
         .listSync()
         .whereType<Directory>()
+        // Get the most recently modified xcarchive to handle cases where an app
+        // may produce multiple xcarchives with different names.
+        // This still could grab the wrong ipa, if multiple `flutter` commands
+        // are running in parallel or the clock is/was broken on the machine.
+        // If either of those occurs in the wild we can check the contents of
+        // the xcarchives, but this should be good enough for now.
```

Comment:
@bryanoltman 
Isn't this crucial?
As the bug in the `shorebird release` itself, this will break patching too not just previewing. Developer will get a confirmation of the patch successfully published but it'll never be installed for app users.

Reply:
It is indeed crucial. We believe that it is a "good enough" fix, in that it will solve the problem in 99.9% of cases, except those outlined in the comment above.

Reply:
Understood. Thanks!